pkg/controller/restore_finalizer_controller.go

```diff
@@ -304,6 +305,24 @@ func (ctx *finalizerContext) patchDynamicPVWithVolumeInfo() (errs results.Result
 						return false, err
 					}
 
+					// check if pvc is not nil and has a pending status
```

Comment:
Could you add more information about this scenario in the comment?
This fix is for a corner case, and it's better to give more information to understand what this does.

Reply:
@blackpiglet I agree that the comment should be clearer on when this happens, but it's not a corner case. WaitForFirstConsumer is a pretty common configuration -- this happens any time you have a WaitForFirstConsumer PVC that's not mounted by a pod. Lots of people are going to hit this.

Reply:
Done, updated the PR.